Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -256,24 +256,26 @@ void ABenchmarkGymGameModeBase::Tick(float DeltaSeconds)
 		PrintMetricsTimer.SetTimer(10);
 	}
 #if	STATS
-	if (StatStartFileTimer.HasTimerGoneOff())
+	if (CPUProfileInterval > 0 && StatStartFileTimer.HasTimerGoneOff())
 	{
 		FString Cmd(TEXT("stat startfile"));
 		if (GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking())
 		{
 			USpatialNetDriver* SpatialDriver = Cast<USpatialNetDriver>(GetNetDriver());
 			if (ensure(SpatialDriver != nullptr))
 			{
-				Cmd = FString::Printf(TEXT("stat startfile %s.ue4stats"), *SpatialDriver->Connection->GetWorkerId());
+				FString InFileName = FString::Printf(TEXT("%s-%s"), *SpatialDriver->Connection->GetWorkerId(), *FDateTime::Now().ToString(TEXT("%m.%d-%H.%M.%S")));
+				const FString Filename = CreateProfileFilename(InFileName, TEXT(".ue4stats"), true);
+				Cmd.Append(FString::Printf(TEXT(" %s"), *Filename));
 			}
 		}
 		GEngine->Exec(GetWorld(), *Cmd);
-		StatStartFileTimer.SetTimer(999999);
+		StatStartFileTimer.SetTimer(CPUProfileInterval);
 	}
-	if (StatStopFileTimer.HasTimerGoneOff())
+	if (CPUProfileInterval > 0 && StatStopFileTimer.HasTimerGoneOff())
```

Comment:
I prefer to use
if CPUProfileInternal > 0:
{
    if (StatStartFileTimer.HasTimerGoneOff){}
    if (StatStopFileTimer.HasTimerGoneOff){}
}

Reply:
Make sense. Updated.